rs/execution_environment/benches/run-benchmark.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh -eu
+##
+## Helper script to run the specified `BENCH`
+## and store the best (min) results in the `MIN_FILE`.
+##
+
+DEPENDENCIES="awk bash bazel rg sed tail tee"
+which ${DEPENDENCIES} >/dev/null || (echo "Error checking dependencies: ${DEPENDENCIES}" && exit 1)
+
+QUICK="${QUICK:-no}"
+[ "${QUICK}" = "no" ] || BENCH_ARGS="--quick --output-format=bencher"
+
+printf "    %-12s := %s\n" \
+    "BENCH" "${BENCH:?Usage: BENCH='//rs/embedders:heap_bench' ${0}}" \
+    "BENCH_ARGS" "${BENCH_ARGS:=--warm-up-time=1 --measurement-time=1 --output-format=bencher}" \
+    "FILTER" "${FILTER:=}" \
+    "MIN_FILE" "${MIN_FILE:=${0##*/}.min}" \
+    "LOG_FILE" "${LOG_FILE:=${MIN_FILE%.*}.log}"
+
+TMP_FILE="${TMP_FILE:-${MIN_FILE%.*}.tmp}"
+
+bash -c "set -o pipefail; \
+    bazel run '${BENCH}' -- ${FILTER} ${BENCH_ARGS} \
+        2>&1 | tee '${LOG_FILE}' | rg '^(test .* )?bench:' --line-buffered \
+        | sed -uEe 's/^test (.+) ... bench: +/    > bench: \1 /' -Ee 's/^bench: +/    > quick: /'" \
+    || (
+        echo "Error running the benchmark:"
+        tail -10 "${LOG_FILE}" | sed 's/^/    ! /'
+        echo "For more details see: ${LOG_FILE}"
+        exit 1
+    )
+
+if ! [ -s "${MIN_FILE}" ]; then
+    echo "    Setting a new baseline in ${MIN_FILE}"
```

Comment:
NIT: `baseline_file` feels better then `min_file`.

Reply:
To avoid confusion with actual baseline files (i.e. files produced on a previous commit), I've modified the text instead.